python/pyspark/sql/connect/plan.py

```diff
@@ -40,13 +42,29 @@ class InputValidationError(Exception):
     pass
 
 
-class LogicalPlan(object):
+class LogicalPlan:
+
+    _lock: Lock = Lock()
+    _nextPlanId: int = 0
 
     INDENT = 2
 
     def __init__(self, child: Optional["LogicalPlan"]) -> None:
         self._child = child
 
+        plan_id: Optional[int] = None
+        with LogicalPlan._lock:
+            plan_id = LogicalPlan._nextPlanId
+            LogicalPlan._nextPlanId += 1
+
+        assert plan_id is not None
+        self._plan_id = plan_id
+
+    def _create_proto_with_plan_id(self) -> proto.Relation:
```

Comment:
My suggestion is to use a more common name when we add other default parameters captured from the plan object.

```suggestion
    def _create_proto_relation(self) -> proto.Relation:
```

Reply:
done